mimic/test/test_customer.py

```diff
@@ -0,0 +1,116 @@
+import json
+
+from twisted.trial.unittest import SynchronousTestCase
+from twisted.internet.task import Clock
+
+from mimic.core import MimicCore
+from mimic.resource import MimicRoot
+from mimic.test.helpers import json_request, request
+
+
+class CustomerAPITests(SynchronousTestCase):
+
+    """
+    Tests for the Customer API
+    """
+
+    def setUp(self):
+        """
+        Initialize core and root.
+
+        `default_contacts` is the default contact set for any tenant,
+        when the contacts are not set using ``POST /contacts``.
+        """
+        self.core = MimicCore(Clock(), [])
+        self.root = MimicRoot(self.core).app.resource()
+        self.default_contacts = [{"email": "example@example.com", "role": "TECHNICAL"},
+                                 {"email": "example2@example.com", "role": "TECHNICAL"}]
+
+    def get_contacts(self, root, tenant):
+        """
+        Get contacts and verify that the responce code was 200.
+        Returns the contacts list
+        """
+        (response, content) = self.successResultOf(json_request(
+            self, root, "GET",
+            "/v1/customer_accounts/CLOUD/{0}/contacts".format(tenant)))
+        self.assertEqual(200, response.code)
+        return content
+
+    def validate_contact_list(self, response, expected_contacts):
+        """
+        Validate that contacts in the `response` is same as the
+        `expected_contacts`
+        """
+        self.assertEqual(len(response["contact"]), len(expected_contacts))
+        contact_emails = [each_contact["emailAddresses"]["emailAddress"][0]["address"]
+                          for each_contact in response["contact"]]
+        self.assertTrue(expected_contacts[0]["email"] in contact_emails)
+        self.assertTrue(expected_contacts[1]["email"] in contact_emails)
+        contact_roles = [each_contact["roles"]["role"][0]
+                         for each_contact in response["contact"]]
+        self.assertTrue(expected_contacts[0]["role"] in contact_roles)
+        self.assertTrue(expected_contacts[1]["role"] in contact_roles)
+
+    def test_get_default_contacts_for_a_tenant_successfully(self):
+        """
+        The ``GET /contacts`` call returns a list of default contacts
+        for a tenant with the response code 200.
+        """
+        response = self.get_contacts(self.root, "111111")
+        self.validate_contact_list(response, self.default_contacts)
+
+    def test_get_default_contacts_for_multiple_tenants_successfully(self):
+        """
+        The ``GET /contacts`` call returns a list of default contacts
+        for multiple tenants with the response code 200.
```

Comment:
This has been a while, and I don't remember I thought to do this. But I am guessing I wanted to make sure the defaults was set on all tenants and not just the first get contact call.